### ISSUE TITLE
Router extends properties in the wrong order

### ISSUE THREAD

**Issue by maskaro** [chance-an/backbone-mvc#8]
Hi there,

First, thank you for your work, it's very handy!

Second, I've been using your code for a little while and I discovered a minor issue which affects the Router when trying to override any of the default (_inherentRouterProperties).

On line 266 ( inside function extend(){} ) I found this:

return Backbone.Router.extend(_.extend(_inherentRouterProperties, properties, { routes: _routes }));

to work better and as expected than this original line:

return Backbone.Router.extend(_.extend(properties, _inherentRouterProperties, { routes: _routes }));

The original way caused the _inherentRouterProperties to resist overriding, such as the 404 method.

Thank you again!

**Comment by chance-an** [comment]
Hi thanks for reaching! And you are welcome!

Your suggestion is very valid. I will try to work on a fix. Presumably:

```
_.extend({}, _inherentRouterProperties, properties, { routes: _routes }) 
```

**Comment by maskaro** [comment]
Yes, that is what I ended up doing on my local copy, it fixed it and worked great!

function extend(properties) {
                var _routes = _.extend(properties.routes || {}, _inherentRouterProperties.routes );
                return Backbone.Router.extend(_.extend(_inherentRouterProperties, properties, { routes: _routes }));
            }

All the best to you!

**Comment by chance-an** [comment]
Underscore method [_.extend(dest, source)](http://underscorejs.org/#extend) modifies `dest`, so you might want to do

```
_.extend({}, _inherentRouterProperties, properties, ...) 
```
to avoid polluting `_inherentRouterProperties` or getting any other potential side effect.

**Comment by maskaro** [comment]
Ah, good to know, thank you!

Take care.

**Comment by chance-an** [comment]
Fixed in pull request [#9](https://github.com/chance-an/backbone-mvc/pull/9). Thanks for reporting this issue! Take care too!
